test/lit.cfg

```diff
@@ -1340,6 +1340,76 @@ elif run_os == 'linux-androideabi' or run_os == 'linux-android':
         '-L%s' % make_path(test_resource_dir, config.target_sdk_name)])
     # The Swift interpreter is not available when targeting Android.
     config.available_features.discard('swift_interpreter')
+elif run_os == 'wasi':
+    tools_directory = pipes.quote(make_path(config.wasi_sdk_path, "bin"))
+
+    lit_config.note("Testing WebAssembly/WASI " + config.variant_triple)
+
+    config.target_object_format = "wasm"
+    config.target_shared_library_prefix = 'lib'
+    config.target_shared_library_suffix = ".a"
+    config.target_sdk_name = "wasi"
+    config.target_runtime = "native"
+
+    config.target_swift_autolink_extract = inferSwiftBinary("swift-autolink-extract")
+
+    config.target_build_swift = ' '.join([
+        '%s', '-target %s',
+	'-Xcc --sysroot=%s', '-Xclang-linker --sysroot=%s',
+	'-tools-directory %s',
+	'-toolchain-stdlib-rpath %s', '%s %s %s %s'
+        ]) % (config.swiftc, config.variant_triple,
+	   config.variant_sdk, config.variant_sdk,
+	   tools_directory, resource_dir_opt,
+	   mcp_opt, config.swift_test_options,
+	   config.swift_driver_test_options, swift_execution_tests_extra_flags)
+    config.target_codesign = "echo"
+    config.target_build_swift_dylib = (
+        "%s -parse-as-library -emit-library -o '\\1'"
+        % (config.target_build_swift))
+    config.target_add_rpath = ''
+    config.target_swift_frontend = ' '.join([
+        '%s', '-frontend', '-target %s',
+	'-Xcc --sysroot=%s', '-tools-directory %s',
+        '%s %s %s %s ',
+        ]) % (config.swift, config.variant_triple,
+           config.variant_sdk, tools_directory,
+           resource_dir_opt, mcp_opt,
+           config.swift_test_options, config.swift_frontend_test_options)
+    subst_target_swift_frontend_mock_sdk = config.target_swift_frontend
+    subst_target_swift_frontend_mock_sdk_after = ""
+    config.target_run = 'wasmer run --backend cranelift --'
```

Comment:
Please use the absolute path to the `wasmer` binary by finding the tool (and logging the location)